src/nodedef.cpp

```diff
@@ -683,9 +683,42 @@ static void fillTileAttribs(ITextureSource *tsrc, TileLayer *layer,
 		}
 	}
 }
-#endif
 
-#ifndef SERVER
+bool ContentFeatures::textureAlphaCheck(ITextureSource *tsrc, const TileDef *tiles, int length)
+{
+	video::IVideoDriver *driver = RenderingEngine::get_video_driver();
+	std::set<std::string> seen;
+	for (int i = 0; i < length; i++) {
+		if (seen.find(tiles[i].name) != seen.end())
+			continue;
+		seen.insert(tiles[i].name);
+
+		// Load the texture and see if there's any transparent pixels
+		video::ITexture *texture = tsrc->getTexture(tiles[i].name);
+		video::IImage *image = driver->createImage(texture,
+			core::position2d<s32>(0, 0), texture->getOriginalSize());
+		core::dimension2d<u32> dim = image->getDimension();
+		bool ok = true;
+		for (u16 x = 0; x < dim.Width; x++) {
+			for (u16 y = 0; y < dim.Height; y++) {
+				if (image->getPixel(x, y).getAlpha() < 255) {
+					ok = false;
+					break;
+				}
+			}
+		}
+		image->drop();
+		if (!ok) {
+			warningstream << "Texture \"" << tiles[i].name << "\" of \""
+				<< name << "\" has transparent pixels, assuming "
+				"use_texture_alpha = true. You should update "
+				"the nodedef!" << std::endl;
```

Comment:
Surely it is far more likely this warning will occur due to a a nodebox that is intended to be opaque but which has transparent pixels used for the 'dead space' in the texture? For example MTG default:mese_post_light.

This warning suggests that 'use_texture_alpha' should be set to 'true', but that is the worng thing to do in many or most cases.
So it seems the warning should advise one of 2 actions:
* If the nodebox/mesh is intended to have transparency, set 'use_texture_alpha = true'.
* Otherwise, remove all transparent pixels from the textures.

Reply:
Almost all warnings I got from MTG were *not* false-positives (as far as I could see).
I didn't even think of the case where "dead" space is filled with transparency and there's no reasonable way this could be detected.

> Otherwise, remove all transparent pixels from the textures.

This is overall a good idea since it both resolves this edge case and potentially allows more texture compression, but I don't know if we should recommend this practice.
But you're right that the error message needs some adjusting to cover this case.

I had also hoped to remove this compatibility code after one (or two?) releases, so a more permanent solution to get rid of the warnings won't be needed.

Reply:
Last time I checked, use_texture_alpha was not the texture's property. So if you strip the alpha, you have to do it to a copy of the texture. Otherwise you'll break the nodes and entities that _do_ use alpha from the same texture.

Reply:
Adjusted the warning message.

Reply:
Message looks good.